File: mao-ai-code-mother-frontend/src/api/materialController.ts
// @ts-ignore
// API 更新时间：
// API 唯一标识：
import request from '@/request'

/** 上传素材（保存到 COS 并落库） POST /material/upload */
export async function uploadMaterial(body: FormData, options?: { [key: string]: any }) {
  return request<API.BaseResponseString>('/material/upload', {
    method: 'POST',
    data: body,
    ...(options || {}),
  })
}

/** 删除素材 POST /material/delete */
export async function deleteMaterial(body: API.DeleteRequest, options?: { [key: string]: any }) {
  return request<API.BaseResponseBoolean>('/material/delete', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

/** 根据 id 获取素材详情（公开） GET /material/get/vo */
export async function getMaterialVOById(params: { id: number | string }, options?: { [key: string]: any }) {
  return request<API.BaseResponseMaterialVO>('/material/get/vo', {
    method: 'GET',
    params: {
      ...params,
    },
    ...(options || {}),
  })
}

/** 分页获取我的素材列表 POST /material/list/page/vo */
export async function listMyMaterialVoByPage(body: API.MaterialQueryRequest, options?: { [key: string]: any }) {
  return request<API.BaseResponsePageMaterialVO>('/material/list/page/vo', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

/** 新建文件夹 POST /material/folder/add */
export async function addMaterialFolder(body: API.MaterialFolderAddRequest, options?: { [key: string]: any }) {
  return request<API.BaseResponseString>('/material/folder/add', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

/** 查询当前用户的文件夹列表 GET /material/folder/list */
export async function listMaterialFolder(options?: { [key: string]: any }) {
  return request<API.BaseResponseListMaterialFolderVO>('/material/folder/list', {
    method: 'GET',
    ...(options || {}),
  })
}

/** 删除文件夹（素材不会被删除，仅解除归属） POST /material/folder/delete */
export async function deleteMaterialFolder(body: API.DeleteRequest, options?: { [key: string]: any }) {
  return request<API.BaseResponseBoolean>('/material/folder/delete', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

File: src/main/java/com/mao/maocodemother/controller/MaterialController.java
package com.mao.maocodemother.controller;

import com.mybatisflex.core.paginate.Page;
import com.mao.maocodemother.common.BaseResponse;
import com.mao.maocodemother.common.DeleteRequest;
import com.mao.maocodemother.common.ResultUtils;
import com.mao.maocodemother.exception.ErrorCode;
import com.mao.maocodemother.exception.ThrowUtils;
import com.mao.maocodemother.model.dto.material.MaterialFolderAddRequest;
import com.mao.maocodemother.model.dto.material.MaterialQueryRequest;
import com.mao.maocodemother.model.entity.User;
import com.mao.maocodemother.model.vo.MaterialFolderVO;
import com.mao.maocodemother.model.vo.MaterialVO;
import com.mao.maocodemother.service.MaterialFolderService;
import com.mao.maocodemother.service.MaterialService;
import com.mao.maocodemother.service.UserService;
import jakarta.annotation.Resource;
import jakarta.servlet.http.HttpServletRequest;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RequestPart;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

/**
 * 素材库接口（上传到 COS 并落库 / 我的素材分页 / 文件夹 CRUD）
 */
@RestController
@RequestMapping("/material")
public class MaterialController {

    @Resource
    private MaterialService materialService;

    @Resource
    private MaterialFolderService materialFolderService;

    @Resource
    private UserService userService;

    /**
     * 上传素材（multipart/form-data：file 必填，name/folderId 可选）
     */
    @PostMapping("/upload")
    public BaseResponse<String> uploadMaterial(@RequestPart("file") MultipartFile file,
                                             @RequestParam(value = "name", required = false) String name,
                                             @RequestParam(value = "folderId", required = false) Long folderId,
                                             HttpServletRequest request) {
        User loginUser = userService.getLoginUser(request);
        Long id = materialService.uploadMaterial(file, name, folderId, loginUser.getId());
        // 雪花 ID 超过 JS 安全整数范围，转字符串返回避免前端精度丢失
        return ResultUtils.success(String.valueOf(id));
    }

    /**
     * 删除素材
     */
    @PostMapping("/delete")
    public BaseResponse<Boolean> deleteMaterial(@RequestBody DeleteRequest deleteRequest,
                                                HttpServletRequest request) {
        ThrowUtils.throwIf(deleteRequest == null || deleteRequest.getId() == null,
                ErrorCode.PARAMS_ERROR, "素材 id 不能为空");
        User loginUser = userService.getLoginUser(request);
        return ResultUtils.success(materialService.deleteMaterial(deleteRequest.getId(), loginUser.getId()));
    }

    /**
     * 查询素材详情（本人或管理员）
     */
    @GetMapping("/get/vo")
    public BaseResponse<MaterialVO> getMaterialVOById(@RequestParam Long id,
                                                      HttpServletRequest request) {
        User loginUser = userService.getLoginUser(request);
        return ResultUtils.success(materialService.getMaterialVO(id, loginUser.getId()));
    }

    /**
     * 分页查询素材（默认查本人的）
     */
    @PostMapping("/list/page/vo")
    public BaseResponse<Page<MaterialVO>> listMaterialVOByPage(@RequestBody MaterialQueryRequest queryRequest,
                                                               HttpServletRequest request) {
        User loginUser = userService.getLoginUser(request);
        return ResultUtils.success(materialService.listMaterialVOPage(queryRequest, loginUser.getId()));
    }

    /**
     * 新建文件夹
     */
    @PostMapping("/folder/add")
    public BaseResponse<String> addMaterialFolder(@RequestBody MaterialFolderAddRequest addRequest,
                                                HttpServletRequest request) {
        ThrowUtils.throwIf(addRequest == null, ErrorCode.PARAMS_ERROR, "请求参数为空");
        User loginUser = userService.getLoginUser(request);
        Long id = materialFolderService.addFolder(addRequest.getName(), loginUser.getId());
        // 雪花 ID 超过 JS 安全整数范围，转字符串返回避免前端精度丢失
        return ResultUtils.success(String.valueOf(id));
    }

    /**
     * 查询我的文件夹列表
     */
    @GetMapping("/folder/list")
    public BaseResponse<List<MaterialFolderVO>> listMaterialFolder(HttpServletRequest request) {
        User loginUser = userService.getLoginUser(request);
        return ResultUtils.success(materialFolderService.listUserFolders(loginUser.getId()));
    }

    /**
     * 删除文件夹
     */
    @PostMapping("/folder/delete")
    public BaseResponse<Boolean> deleteMaterialFolder(@RequestBody DeleteRequest deleteRequest,
                                                      HttpServletRequest request) {
        ThrowUtils.throwIf(deleteRequest == null || deleteRequest.getId() == null,
                ErrorCode.PARAMS_ERROR, "文件夹 id 不能为空");
        User loginUser = userService.getLoginUser(request);
        return ResultUtils.success(materialFolderService.deleteFolder(deleteRequest.getId(), loginUser.getId()));
    }
}
